packages/flutter_tools/lib/src/asset.dart

```diff
@@ -203,6 +230,24 @@ class _Asset {
 
   @override
   String toString() => 'asset: $assetEntry';
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other)) return true;
+    if (other.runtimeType != runtimeType) return false;
+    final _Asset otherAsset = other;
+    return base == otherAsset.base &&
+        assetEntry == otherAsset.assetEntry &&
+        relativePath == otherAsset.relativePath &&
+        source == otherAsset.source;
+  }
+
+  @override
+  int get hashCode =>
```

Comment:
IIRC, we usually do something fancier to combine the hashcodes (multiply, xor, or shift), and make sure we stay in the 30-bit range.